application/server/server.go

```diff
@@ -91,11 +91,29 @@ func initHandlers(srv *jrpc2.Server, handlers *handler.Map) {
 	(*handlers)["shutdown"] = Shutdown()
 	(*handlers)["exit"] = Exit(srv)
 	(*handlers)["workspace/didChangeWorkspaceFolders"] = WorkspaceDidChangeWorkspaceFoldersHandler()
+	(*handlers)["workspace/willDeleteFiles"] = WorkspaceWillDeleteFilesHandler()
 	(*handlers)["workspace/didChangeConfiguration"] = WorkspaceDidChangeConfiguration(srv)
 	(*handlers)["window/workDoneProgress/cancel"] = WindowWorkDoneProgressCancelHandler()
 	(*handlers)["workspace/executeCommand"] = ExecuteCommandHandler(srv)
 }
 
+// WorkspaceWillDeleteFilesHandler handles the workspace/willDeleteFiles message that's raised by the client
+// when files are deleted
```

Comment:
Why not "workspace/didDeleteFiles", since we are interested in the end-result?

Reply:
I don't think there's a reason to prefer one over the other, but if we remove the diagnostics after the file was removed than there might be a moment in time where the diagnostics are still showing in the IDE even though the files are no longer there, so I preferred this message.

Reply:
Sounds fair. Does LSP4E support `workspace/willDeleteFiles`?

Reply:
It doesn't, but Eclipse removes the errors automatically

Reply:
If by errors you mean reported diagnostics, then 👌